yarn-project/noir-contracts/src/contracts/native_token_contract/src/main.nr

```diff
@@ -355,7 +337,7 @@ contract NativeToken {
     // Note 2: Having it in all the contracts gives us the ability to compute the note hash and nullifier differently for different kind of notes.
     unconstrained fn compute_note_hash_and_nullifier(contract_address: Field, nonce: Field, storage_slot: Field, preimage: [Field; VALUE_NOTE_LEN]) -> [Field; 4] {
         let note_header = NoteHeader { contract_address, nonce, storage_slot };
-        if (storage_slot == 2) {
+        if (storage_slot == 3) {
```

Comment:
Curious how you found this

Reply:
Had failures because of missing `compute_note_hash_and_nullifier` so debugged from there and saw that the slot did not match what was in storage. The actual issue was then the different selectors 😬 leading to the loop.